core/src/test/java/io/hyperfoil/core/session/BaseBenchmarkParserTest.java

```diff
@@ -0,0 +1,34 @@
+package io.hyperfoil.core.session;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+
+import io.hyperfoil.api.config.Benchmark;
+import io.hyperfoil.core.parser.BenchmarkParser;
+import io.hyperfoil.core.parser.ParserException;
+import io.hyperfoil.core.test.TestUtil;
+import io.hyperfoil.impl.Util;
+
+public abstract class BaseBenchmarkParserTest {
+
+   protected Benchmark loadScenario(String name) {
+      try {
+         InputStream config = getClass().getClassLoader().getResourceAsStream(name);
+         Benchmark benchmark = loadBenchmark(config);
+         // Serialization here is solely for the purpose of asserting serializability for all the components
+         byte[] bytes = Util.serialize(benchmark);
+         assertThat(bytes).isNotNull();
```

Comment:
you can rehydrate it back and see if they are equivalent/can work

Reply:
Hey @franz1981, you mean to serialize the Benchmark and then deserialize it checking if it will match the original Benchmark?

Reply:
Yep, in case in the future we would make some of the field `transient` (or deps field); probably is just overworrying on my side and we didn't have done it previously just feel free to ignore my advice :D

Reply:
Makes sense, in this case Ithink we could have two options:
- Implement `equals` in order to define our concept of bencharmk equality and just make the comparison everytime we load a benchmark
- Run all test that loads benchmark two times (one with the original benchamrk and one with the deserialized one)

Any preference?

Reply:
The second one increase by far the tests duration and still won't report what was "wrong" , so probably the former is more fine grain

Reply:
good point, let's keep them separated!

Reply:
Created issue https://github.com/Hyperfoil/Hyperfoil/issues/304 to keep track of this